### ISSUE TITLE
'KSampler Adv. (Efficient)' node doesn't react to change in 'Apply ControlNet (Advanced)' node [or other nodes] unless the seed changes!

### ISSUE THREAD

**Issue by zhoobin021** [jags111/efficiency-nodes-comfyui#146]
![workflow (0)](https://github.com/jags111/efficiency-nodes-comfyui/assets/86180928/8258ab3a-c3b9-4b10-9c29-cbd9ca7f8df0)
![workflow (1)](https://github.com/jags111/efficiency-nodes-comfyui/assets/86180928/61ac38bf-20a4-4000-9086-4def4b47bbc8)
Images have workflow embed inside them. please download them and test.

As you can see in the first image, the strength of 'Apply ControlNet (Advanced)' is zero and the both generate the same image. but in second image the strength is changed to 1 but KSampler Adv. (Efficient) doesn't change the generated image!

Unless you change the seed in 'KSampler Adv. (Efficient)' it doesn't apply any changes before the node that was made, I tried changing 'Load ControlNet Model' & 'Load Image' & preprocessor but nothing changed it unless you change the seed and generate again.

so basically in order to apply the changed you should first change seed once and generate and then move back to the previous seed to generate the desired image with its chosen seed,

**Comment by zos474** [comment]
Would it be possible to at least add a setting on the KSampler (Efficient) that would cause it to always run?  I really love the preview ability but it is such a setback not to be able to keep the seed constant and do things like cfg / controlnet setting variations (they just all spit out the same result)

**Comment by jags111** [comment]
sure this needs to be checked again

**Comment by bobcate** [comment]
Reminder that this is still a thing.

**Comment by flt6** [comment]
Still now

**Comment by mbylstra** [comment]
This explains why "XY Input: Control Net" and "XY Input: Control Net Plot" always output the same image for every cell in the grid when using XYPlot
